File: BFS & DFS/dfs.java
import java.util.HashMap;
import java.util.HashSet;
import java.util.Set;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class DFSExample {

    public static void main(String[] args) {
        Map<String, List<String>> graph = new HashMap<>();
        graph.put("A", new ArrayList<String>(List.of("B", "C")));
        graph.put("B", new ArrayList<String>(List.of("D", "E")));
        graph.put("C", new ArrayList<String>(List.of("F")));
        graph.put("D", new ArrayList<>());
        graph.put("E", new ArrayList<String>(List.of("F")));
        graph.put("F", new ArrayList<>());

        Set<String> visited = new HashSet<>();
        dfs(graph, "D", visited);
    }

    public static void dfs(Map<String, List<String>> graph, String start, Set<String> visited) {
        visited.add(start);
        System.out.println(start);

        for (String neighbor : graph.get(start)) {
            if (!visited.contains(neighbor)) {
                dfs(graph, neighbor, visited);
            }
        }
    }
}

File: BFS & DFS/dfs.py
def dfs(graph, start, visited=None):
    if visited is None:
        visited = set()
    visited.add(start)  

    print(start)
    
    for neighbor in graph[start]:
        if neighbor not in visited:
            dfs(graph, neighbor, visited)
    return visited


graph = {
    'A': ['B', 'C'],
    'B': ['A', 'D', 'E'],
    'C': ['A', 'F' , 'G'],
    'D': ['B'],
    'E': ['B'],
    'F': ['C'],
    'G': ['C']
}

dfs(graph=graph, start="A")